backend/test/features/support/steps.js

```diff
@@ -1,25 +1,37 @@
 // features/support/steps.js
-import { Given, When, Then, AfterAll } from 'cucumber'
+import { Given, When, Then, BeforeAll } from 'cucumber'
 import { expect } from 'chai'
-// import { client } from '../../../src/activitypub/apollo-client'
 import { GraphQLClient } from 'graphql-request'
 import Factory from '../../../src/seed/factories'
 import { host } from '../../../src/jest/helpers'
-const debug = require('debug')('ea:test:steps')
 
 const factory = Factory()
 const client = new GraphQLClient(host)
+// setDefaultTimeout(60 * 1000)
 
-function createUser (slug) {
-  debug(`creating user ${slug}`)
-  return factory.create('User', {
-    name: slug,
-    email: 'example@test.org',
-    password: '1234'
-  })
+async function createUser (slug) {
+  const result = await client.request(`
+    query {
+      User(slug: "${slug}") {
+        slug
+      }
+    }
+  `)
+  if (result.User.length === 0) {
+    return factory.create('User', {
+      name: slug,
+      email: 'example@test.org',
+      password: '1234',
+      actorId: `http://localhost:3000/api/users/${slug}`
+    })
+  }
   // await login({ email: 'example@test.org', password: '1234' })
 }
 
+BeforeAll('Clean Database for fresh test setup', async function () {
```

Comment:
Great! I did not know cucumber has `BeforeAll` similar to `Given` and `When` I thought we have to use hooks in `features/support/`.

Reply:
Yes, it's very handy.